Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1406,6 +1436,133 @@ function Get-CurrentUserDomain
     return $currentUserDomain
 }
 
+function Get-WebDeployParameters{
+    param(
+        [Parameter(Mandatory=$true)]
+        [Hashtable] $discoveryConfig,
+        [Parameter(Mandatory=$true)]
+        [Hashtable] $commonConfig,
+        [Parameter(Mandatory=$true)]
+        [string] $userName,
+        [string] $registrationApiSecret
+    )
+
+    Confirm-DiscoveryConfig -discoveryConfig $discoveryConfig -commonConfig $commonConfig
+    $metaDataConnectionString = Get-MetadataConnectionStringForDiscovery -commonConfig $commonConfig
+    
+    $webDeployParameters = @(
+                                @{
+                                    Name = "IIS Web Application Name";
+                                    Value = "$($discoveryConfig.siteName)/$($discoveryConfig.appName)"
+                                },
+                                @{
+                                    Name = "App Pool Account";
+                                    Value = $userName
+                                },
+                                @{
+                                    Name = "Application Endpoint Address";
+                                    Value = "https://$($commonConfig.webServerDomain)/$($discoveryConfig.appName)"
+                                },
+                                @{
+                                    Name = "Client Environment";
+                                    Value = "$($commonConfig.clientEnvironment)"
+                                },
+                                @{
+                                    Name = "DiscoveryServiceDataContext-Deployment-Deployment Connection String";
+                                    Value = $metaDataConnectionString
+                                },
+                                @{
+                                    Name = "DiscoveryServiceDataContext-Web.config Connection String";
+                                    Value = $metaDataConnectionString
+                                }
+                            )
+
+    if([string]::IsNullOrEmpty($registrationApiSecret) -ne $true) {
+        $webDeployParameters += @{ 
+            Name = "Discovery Service Api Secret"; 
+            Value = $registrationApiSecret 
+        }
+
+        $webDeployParameters += @{
+            Name = "Fabric.Identity URL";
+            Value = "$($commonConfig.IdentityService)"
+        }
+    }
+    
+    return $webDeployParameters
+}
+
+function Confirm-DiscoveryConfig{
```

Comment:
At some point we should probably create something similar for identity/idpss and have that run in the parent coordination script so the script fails early before trying to install with bad values/quiet mode

Reply:
That would require a huge amount of refactoring.  Im not disagreeing.  Perhaps we can add a PBI for a product enhancement.